warehouse/transformer/transformer.go

```diff
@@ -133,35 +134,43 @@ func (t *Transformer) Transform(_ context.Context, clientEvents []types.Transfor
 }
 
 func (t *Transformer) processWarehouseMessage(cache *cache, event *types.TransformerEvent) ([]map[string]any, error) {
-	if err := t.enhanceContextWithSourceDestInfo(event); err != nil {
-		return nil, fmt.Errorf("enhancing context with source and destination info: %w", err)
+	if err := t.checkValidContext(event); err != nil {
+		return nil, fmt.Errorf("checking valid context: %w", err)
 	}
 	t.addMandatoryFields(event)
 	return t.handleEvent(event, cache)
 }
 
-func (t *Transformer) enhanceContextWithSourceDestInfo(event *types.TransformerEvent) error {
+func (t *Transformer) checkValidContext(event *types.TransformerEvent) error {
 	if !t.config.populateSrcDestInfoInContext.Load() {
 		return nil
 	}
-
-	messageContext, ok := event.Message["context"]
-	if !ok || messageContext == nil {
-		messageContext = map[string]any{}
+	contextVal, exists := event.Message["context"]
+	if !exists || contextVal == nil {
+		return nil
 	}
-	messageContextMap, ok := messageContext.(map[string]any)
-	if !ok {
+	if !utils.IsObject(contextVal) {
 		return response.ErrContextNotMap
 	}
-	messageContextMap["sourceId"] = event.Metadata.SourceID
-	messageContextMap["sourceType"] = event.Metadata.SourceType
-	messageContextMap["destinationId"] = event.Metadata.DestinationID
-	messageContextMap["destinationType"] = event.Metadata.DestinationType
-
-	event.Message["context"] = messageContextMap
 	return nil
 }
 
+func (t *Transformer) eventContext(tec *transformEventContext) any {
+	contextVal, exists := tec.event.Message["context"]
+	if !t.config.populateSrcDestInfoInContext.Load() {
```

Comment:
The if check can be evaluated before `contextVal`

Reply:
We are returning return `contextVal` within the if block right.